akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1384,6 +1384,7 @@ private[stream] object Collect {
             if (isAvailable(out)) {
               if (!hasBeenPulled(in)) tryPull(in)
               push(out, elem)
+              if (isClosed(in) && todo == 0) completeStage()
             } else buffer.enqueue(elem)
           case Success(null) =>
```

Comment:
As an aside, what are those clauses? Is it just silently ignoring null results because akka-stream doesn't support it?

Reply:
Yep, it's the same in MapAsync as far as I can see (has test coverage to ignore nulls)